src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py

```diff
@@ -28,76 +28,87 @@ def _flexible_server_create(cmd, client, resource_group_name=None, server_name=N
-            administrator_login, administrator_login_password, version, tags, public_access, assign_identity,
-            tier, high_availability, zone, delegated_subnet_arguments)
-
-        if public_access is not None:
-            if public_access == 'on':
-                start_ip, end_ip = '0.0.0.0', '255.255.255.255'
+        if server_result is None:
+            # If subnet is provided, use that subnet to create the server, else create subnet if public access is not enabled.
+            if subnet_arm_resource_id is not None:
+                subnet_id = subnet_arm_resource_id  # set the subnet id to be the one passed in
+                delegated_subnet_arguments = mysql.flexibleservers.models.DelegatedSubnetArguments(
+                    subnet_arm_resource_id=subnet_id)
+            elif public_access is None and subnet_arm_resource_id is None:
+                subnet_id = create_vnet(cmd, server_name, location, resource_group_name,
+                                        "Microsoft.DBforPostgreSQL/flexibleServers")
+                delegated_subnet_arguments = mysql.flexibleservers.models.DelegatedSubnetArguments(
+                    subnet_arm_resource_id=subnet_id)
             else:
-                start_ip, end_ip = parse_public_access_input(public_access)
-            firewall_id = create_firewall_rule(db_context, cmd, resource_group_name, server_name, start_ip, end_ip)
-
-        # Create mysql database if it does not exist
-        if database_name is None:
-            database_name = DEFAULT_DB_NAME
-        _create_database(db_context, cmd, resource_group_name, server_name, database_name)
-
-    user = server_result.administrator_login
-    id = server_result.id
-    loc = server_result.location
-    version = server_result.version
-    sku = server_result.sku.name
-
-    # This is a workaround for getting the hostname right .
-    # Without Vnet, the hostname returned by the API os <servername>.postgres.database.azure.com
-    # with Vnet enabled, the hostname returned by the API is <servername>.<servername>.postgres.database.azure.com
-    if len(server_result.fully_qualified_domain_name.split('.')) != 6:
+                delegated_subnet_arguments = None
+
+            # Create mysql server
+            # Note : passing public_access has no effect as the accepted values are 'Enabled' and 'Disabled'. So the value ends up being ignored.
+            server_result = _create_server(db_context, cmd, resource_group_name, server_name, location, backup_retention,
+                                           sku_name, tier, storage_mb, administrator_login, administrator_login_password,
+                                           version, tags, delegated_subnet_arguments, assign_identity, public_access,
+                                           high_availability, zone)
+
+            # Adding firewall rule
+            if public_access is not None and public_access != 'none':
```

Comment:
.tolower() needed? If customer passes in something like "NONE", this won't be caught

Reply:
same in the elif

Reply:
Addressed